sentry/src/routes/channel.rs

```diff
@@ -342,6 +336,74 @@ pub async fn get_spender_limits<A: Adapter + 'static>(
     Ok(success_response(serde_json::to_string(&res)?))
 }
 
+pub async fn get_all_spender_limits<A: Adapter + 'static>(
+    req: Request<Body>,
+    app: &Application<A>,
+) -> Result<Response<Body>, ResponseError> {
+    let channel = req
+        .extensions()
+        .get::<ChannelV5>()
+        .expect("Request should have Channel")
+        .to_owned();
+
+    let new_state = match get_corresponding_new_state(&app.pool, &channel).await? {
+        Some(new_state) => new_state,
+        None => return Err(ResponseError::NotFound),
+    };
```

Comment:
Since we need to handle the use case of the Channel (see bellow) we need:

```suggestion
    let new_state = get_corresponding_new_state(&app.pool, &channel).await?; // an `Option<MessageResponse<NewState<CheckedState>>>`
```